pkg/cluster/internal/create/actions/config/config.go

```diff
@@ -176,6 +177,23 @@ func getKubeadmConfig(cfg *config.Cluster, data kubeadm.ConfigData, node nodes.N
 	}
 	data.KubernetesVersion = kubeVersion
 
+	// TODO: gross hack!
```

Comment:
this is basically the same gross hack, and now somewhat orthogonal to this PR, but it is slightly more robust now ...